### ISSUE TITLE
PreFetch / PostFetch callbacks

### ISSUE THREAD

**Issue by FabianTe** [bitinn/node-fetch#483]
Hi!

TLDR: I would like to be able to call my code (via callback) before and after requests, especially on **redirect** responses.

Currently I am using this package to create automated tests that call some microservices. Those are secured and the user has to be authorized via a cookie. For this I am using `tough-cookie` which works fine by itself. Am am using it to insert cookie headers before `fetch` requests and it is storing received cookies from `Response` instances via the `set-cookie` header.

This does not work for server responses that want the client to redirect. `node-fetch` is able to follow redirects, yes, but I am not able to store cookies from the pages that were received while in the redirect loop.

Using redirect manually is just such a hassle. Hence I though that it would be useful to have some callbacks which enable me to do X before **every** request - manipulating the request parameters / setting cookies - and to do Y after **every** response was received. Important thing, this would have to apply to redirects.

Any thoughts?

**Comment by TimothyGu** [comment]
See https://github.com/bitinn/node-fetch/issues/94#issuecomment-201700557.

**Comment by FabianTe** [comment]
Yeah, I am now using fetch-cookie for that stuff.
